src/os/bluestore/BlueStore.cc

```diff
@@ -3885,6 +3885,8 @@ int BlueStore::BlueReadTrans::_do_read(
     // yes, at least one region isn't in cache
     if (!aio) {
       aio = std::make_unique<AioReadBatch>(store->cct);
+      Collection *c = static_cast<Collection *>(this->c.get());
```

Comment:
FIXME: this is just clumsy. The hint should be private and set with constructor's assistance.